tests/keras/metrics_test.py

```diff
@@ -116,13 +116,12 @@ class BinaryTruePositives(keras.layers.Layer):
         Assumes predictions and targets of shape `(samples, 1)`.
 
         # Arguments
-            threshold: Float, lower limit on prediction value that counts as a
-                positive class prediction.
             name: String, name for the metric.
         """
 
         def __init__(self, name='true_positives', **kwargs):
             super(BinaryTruePositives, self).__init__(name=name, **kwargs)
+            self.stateful = True
```

Comment:
Would we have to specify this attribute for each metric?

Reply:
That's the idea, indeed - as long as your metric behaves as a stateful layer.